be/src/product/dto/update-product.input.ts

```diff
@@ -0,0 +1,19 @@
+import { CreateProductInput } from './create-product.input';
+import { InputType, Field, Int, PartialType } from '@nestjs/graphql';
+
+@InputType()
+export class UpdateProductInput extends PartialType(CreateProductInput) {
+  @Field(() => Int)
+  id: number;
```

Comment:
You don't need to re-write to properties for the class if the class extends the `CreateProductInput`.
[partial doc](https://docs.nestjs.com/openapi/mapped-types#partial)